cmd/deploy/waiter.go

```diff
@@ -0,0 +1,87 @@
+package deploy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/armory/armory-cli/internal/clierr"
+	"github.com/armory/armory-cli/internal/clierr/exitcodes"
+	"github.com/armory/armory-cli/internal/graphql"
+	"github.com/armory/armory-cli/pkg/console"
+	"go.uber.org/zap"
+	"time"
+)
+
+const (
+	defaultWaiterTickDuration = 5 * time.Second
+)
+
+type (
+	Waiter struct {
+		client              pipelineClient
+		tick                time.Duration
+		log                 *zap.SugaredLogger
+		cloudConsoleBaseURL string
+	}
+
+	pipelineClient interface {
+		GetPipeline(ctx context.Context, pipelineID string) (*graphql.Pipeline, error)
+	}
+)
+
+func NewWaiter(
+	gqlclient *graphql.Client,
+	cloudConsoleBaseURL string,
+) *Waiter {
+	return &Waiter{
+		client:              gqlclient,
+		tick:                defaultWaiterTickDuration,
+		cloudConsoleBaseURL: cloudConsoleBaseURL,
+	}
+}
+
+func (w *Waiter) WaitForPipelineToBeProcessed(ctx context.Context, pipelineID string) error {
```

Comment:
why bother with this custom implementation and not use stuff from our go-commons awaitability package?